packages/charts/react-charting/src/components/DeclarativeChart/PlotlySchemaAdapter.ts

```diff
@@ -1257,3 +1340,43 @@ const getLegendProps = (data: Data[], layout: Partial<Layout> | undefined) => {
       layout?.showlegend === false || (layout?.showlegend !== true && legends.length < 2) ? true : hideLegends,
   };
 };
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const isInvalidValue = (value: any) => {
+  return typeof value === 'undefined' || value === null || (typeof value === 'number' && !isFinite(value));
+};
+
+const getNumberAtIndexOrDefault = (data: PlotData['z'] | undefined, index: number) => {
+  if (isArrayOrTypedArray(data)) {
+    if (typeof data![index] !== 'number' || !isFinite(data![index] as number)) {
+      return;
+    }
+
+    return data![index] as number;
+  }
+
+  return 1;
+};
+
+export const getValidXYRanges = (series: Partial<PlotData>) => {
```

Comment:
where are you handling the case of tan(X) where y was coming 1k and the range was set as -10 to 10

Reply:
Discussed in meeting to log a bug for it. Should I handle it here?

Reply:
I forgot what was the conclusion. We can take it separately.

Reply:
logged a bug for it